src/csl/initSanitizer.h

```diff
@@ -93,13 +108,57 @@ class InitSanitizer {
     {
         m_safe = false;
     }
+    
+    bool operator == (const InitSanitizer<T> &other) const
+    {
+      return  (hasValue() && other.hasValue()) ?  (static_cast<T>(get())==static_cast<T>(other.get())) : (!hasValue() && !other.hasValue()) ;
+    }
+    
+    bool operator == (const T &other) const
+    {
+      return  hasValue() ? (m_value==other) : false;
+    }
 
-  public:
-    char const *name = "unnamed_var";
+    bool operator != (InitSanitizer<T> &other) const
+    {
+      return  !(*this == other) ;
+    }
+    
+    bool operator != (T &other) const
+    {
+      return  !(*this == other) ;
+    }
+    
+    
+    operator T() const
+    {
+        return get();
+    }
 
+    std::string getName() const 
+    {
+      return name;
+    }
+    
+    void setName(const std::string value)
```

Comment:
The best way to write a function like this is the following:
``` cpp
void setName(std::string value)
{
    name = std::move(value);
}
```
In this way the string is not copied inside the function, and passing it by copy allows to create only one string when doing for example `param.setName("m_H")`. Otherwise, you can simply pass the string by reference adding the `&`: `std::string const &value`.